server/src/main/java/org/apache/iotdb/db/newsync/sender/service/SenderService.java

```diff
@@ -175,35 +181,45 @@ public Pipe parseCreatePipePlan(CreatePipePlan plan, PipeSink pipeSink, long pip
       }
     }
 
-    // get TsFilePipe
-    //    PipeSink.Type pipeSinkType = pipeSink.getType();
-    //    if (!pipeSinkType.equals(PipeSink.Type.IoTDB)) {
-    //      throw new PipeException(
-    //          String.format(
-    //              "Wrong pipeSink type %s for create TsFilePipe.", pipeSinkType)); // internal
-    // error
-    //    }
-    return new TsFilePipe(
-        pipeCreateTime, plan.getPipeName(), pipeSink, plan.getDataStartTimestamp(), syncDelOp);
+    TsFilePipe pipe =
+        new TsFilePipe(
+            pipeCreateTime, plan.getPipeName(), pipeSink, plan.getDataStartTimestamp(), syncDelOp);
+    try {
+      if (!(pipeSink instanceof IoTDBPipeSink)) {
+        throw new PipeException(
+            String.format(
+                "Wrong pipeSink type %s for create pipe %s", pipeSink.getType(), pipe.getName()));
+      }
+      ITransportClient transportClient =
+          new TransportClient(
+              pipe, ((IoTDBPipeSink) pipeSink).getIp(), ((IoTDBPipeSink) pipeSink).getPort());
+      pipe.setTransportHandler(
```

Comment:
It's neccessary to send heartbeat CREATE to receiver to regisiter this pipe.

Why not add a new function `create()` for pipe, which creates  TransportClient and send CREATE heartbeat?